### PR TITLE
create popup

### PR DESCRIPTION
**Pull request by Pozzitive11**
### HTML-CSS-Popup

[Demo](https://pozzitive11.github.io/pure-popup/)
[Code base](https://github.com/Pozzitive11/pure-popup)

The code is submitted in a dedicated feature branch. Please, review.

### CODE REVIEW

**Comment by artem-trubin** [comment]
Hi! Please host your project on Github Pages in a separate repository and post a link to it in your PR description.

**Comment by Pozzitive11** [comment]
https://pozzitive11.github.io/pure-popup/

**Comment by A-Ostrovnyy** [comment]
@Pozzitive11 please edit your first message according to [template](https://github.com/kottans/frontend-2022-homeworks/blob/main/.github/pull_request_template.md)
Also, all next homework create with this template.

**Comment by Pozzitive11** [comment]
[Demo](https://pozzitive11.github.io/pure-popup/)
[Code base](https://github.com/Pozzitive11/pure-popup)

**Comment by artem-trubin** [comment]
Nice! There are some things to point out.
1. Add Demo and Code base links to the PR description, not just as a comment.
2. Try to walk through your buttons with keyboard. You'll see that it targets the popup button, but you cannot open it with a keyboard. THEN, you tab to an invisible object, and then you can open a popup with a key press. It happens because your label and checkbox work as separate buttons, fix this. The same goes for the "More" button.
3. Some of your icons seem stretched, please fix this too.

File: submissions/pozzitive11/popup/css/style.css [hunk 1]
```diff
@@ -0,0 +1,215 @@
+*,
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/css/style.css:1]: :fire:

File: submissions/pozzitive11/popup/css/style.css [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  border-radius: 30px;
+}
+
+.nav__list {
+  display: flex;
+  align-items: center;
+  justify-content: flex-end;
+}
+
+.nav__item:not(:last-child) {
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/css/style.css:50]: :fire:

File: submissions/pozzitive11/popup/css/style.css [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+.nav__link:hover,
+.nav__link:focus {
+  text-decoration: underline;
+}
+
+.nav__btn:hover,
+.nav__btn:focus {
+  background-color: #ddd;
+}
+
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/css/style.css:71]: Avoid unused extra lines

File: submissions/pozzitive11/popup/css/style.css [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+.popup__img {
+  width: 26px;
+  height: 26px;
+  cursor: pointer;
+}
+
+.popup__input {
+  appearance: none;
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/css/style.css:97]: :fire:

File: submissions/pozzitive11/popup/css/style.css [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+  display: block;
+}
+
+.menu__extra-input:checked + .menu__extra-btn {
+  display: none;
+}
+
+.nav__item-btn{
+  display: flex;
+}
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/css/style.css:215]: Add empty lines at the end of files. [Reason](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline?noredirect=1&lq=1)

File: submissions/pozzitive11/popup/index.html [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+          </li>
+          <li class="nav__item">
+            <a href="#" class="nav__link">
+              Images
+            </a>
+          </li>
+          <li class="nav__item nav__item-btn">
+            <input id="popup" class="popup__input" type="checkbox">
+            <label class="nav__btn popup__btn" for="popup">
+              <img class="popup__img" src="./images/popup-button.png" alt="popup">
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/index.html:34]: Fix styles on focus state 
image

File: submissions/pozzitive11/popup/index.html [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+            </label>
+            <div class="popup__menu">
+              <div class="menu__main">
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/index.html:37]: Logically it looks like a list, nested list makes it easier to understand navigation structure.

File: submissions/pozzitive11/popup/index.html [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+                  <a href="https://github.com/kottans/frontend" class="menu__extra-link">
+                    Even more from Kottans
+                  </a>
+                </div>
+              </div>
+            </div>
+          </li>
+          <li class="nav__item">
+            <a href="#" class="nav__btn">
+              <img class="avatar" src="images/bell.png" alt="Notification icon">
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/index.html:122]: For decorative images, like icons, not necessarily populate `alt` attribute? better left it empty

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!
See my comment below.

File: submissions/pozzitive11/popup/css/style.css (hunk 1 above)
Comment (Pozzitive11) [reply submissions/pozzitive11/popup/css/style.css:1]: what does fire mean?)

Comment (A-Ostrovnyy) [reply submissions/pozzitive11/popup/css/style.css:1]: It means coll solution).

Comment (Pozzitive11) [reply submissions/pozzitive11/popup/css/style.css:1]: Thx :D

File: submissions/pozzitive11/popup/index.html (hunk 6 above)
Comment (Pozzitive11) [reply submissions/pozzitive11/popup/index.html:34]: maybe there are some tips how to fix it? 🥲

Comment (A-Ostrovnyy) [reply submissions/pozzitive11/popup/index.html:34]: In the browser by default, only interactive elements have focus, In your case focus goes to `input type="checkbox"` but you hide him using `appearance: none`. Find a way to highlight visible element when input on focus.

Comment (Pozzitive11) [reply submissions/pozzitive11/popup/index.html:34]: I want to understand the problem more specifically. That is, the selection of the button should disappear after clicking on it?
image

Comment (Pozzitive11) [reply submissions/pozzitive11/popup/index.html:34]: or in the sense of adding to the checkbox styles `position: absolute; top: 0px; left: 0px; opacity: 0`?

Comment (A-Ostrovnyy) [reply submissions/pozzitive11/popup/index.html:34]: Watch this [video](https://www.youtube.com/watch?v=E6kLaaQFctU&t=413s) for a better understanding of how and why we hide checkbox.)

File: submissions/pozzitive11/popup/index.html [hunk 9]
```diff
@@ -133,4 +169,5 @@
   </header>
 </body>
 
-</html>
\ No newline at end of file
+</html>
+
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/index.html:173]: Avoid unused extralines.

File: submissions/pozzitive11/popup/index.html [hunk 10]
```diff
@@ -29,97 +29,133 @@
...
-                </div>
-              </div>
+                </ul>
+              </ul>
             </div>
           </li>
           <li class="nav__item">
             <a href="#" class="nav__btn">
-              <img class="avatar" src="images/bell.png" alt="Notification icon">
+              <img class="avatar" src="images/bell.png" alt="">
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/index.html:158]: You set `alt` attribute with an empty value for this icon, what about other icons, or do you have only one icon in your popup?)

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Awesome!
Don't forget to update demo page, for the ability to validate your changes.

**Comment by Pozzitive11** [comment]
oops, I named the commit "fix # 2", and github thought I was referencing to another PR

File: submissions/pozzitive11/popup/index.html (hunk 6 above)
Comment (A-Ostrovnyy) [reply submissions/pozzitive11/popup/index.html:34]: Some of your changes brokes styles for focus state. Test your solution before pushing it to the repo.
image

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Rest your icons have filled `alt` attribute. Try to keep your code style consistent: If you remove some attribute/method for one entity why you don't do this for the rest of similar entities?

**Comment by Pozzitive11** [comment]
added some styling for proper display, please check

File: submissions/pozzitive11/popup/css/style.css [hunk 11]
```diff
@@ -208,5 +212,6 @@ li {
 }
 
 .nav__item-btn {
+  /* position: relative; */
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/css/style.css:215]: Avoid committing commented code.

File: submissions/pozzitive11/popup/css/style.css [hunk 12]
```diff
@@ -94,6 +94,10 @@ li {
   -moz-appearance: none;
   appearance: none;
   position: absolute;
+  top: 0;
+  left: 0;
+  cursor: pointer;
+  outline: none;
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/css/style.css:100]: On Google Chrome you can't see it but Firefox shows all) Check this video and see what's wrong https://share.vidyard.com/watch/yLXJRZFy2TUJCFUWRMtnzd?

File: submissions/pozzitive11/popup/index.html [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+                    </li>
+                    <li class="popup__item">
+                      <a class="popup__link" href="#">
+                        <img src="images/files.png" alt="Files">
+                        <p class="popup__item-text">Files</p>
+                      </a>
+                    </li>
+                    <li class="popup__item">
+                      <a class="popup__link" href="#">
+                        <img src="images/contacts.png" alt="Contacts">
```
Comment (A-Ostrovnyy) [inline submissions/pozzitive11/popup/index.html]: Why do you leave the filled `alt` attribute for all these icons, if you remove it in previous commits?

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Read carefully what the mentor writes to you. Check all issues before re-request review.

File: submissions/pozzitive11/popup/index.html (hunk 13 above)
Comment (Pozzitive11) [reply submissions/pozzitive11/popup/index.html]: when the `img` tag is used, isn't an `alt `required?

Comment (A-Ostrovnyy) [reply submissions/pozzitive11/popup/index.html]: Sorry for russians but it's a good [article](https://htmlacademy.ru/blog/articles/alt-text)

Comment (Pozzitive11) [reply submissions/pozzitive11/popup/index.html]: that is, these images are considered _"icons in a button"_ and should not write something in the `alt` tag for them

**Comment by A-Ostrovnyy** [comment]
Well done
@Pozzitive11 it's okay to make mistakes at the beginning of your path to becoming a developer, don't worry about that)
After you tested your changes and pushed them click re-request review button to notify the reviewer about finishing your task.
image

**Pull request merged**